metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1988,6 +1990,26 @@ def _build_pre_aggregation_plan(
                 parent_node=output_node, time_dimension_spec=custom_granularity_spec
             )
 
+        # Filter to specs needed for the rest of the query. This is needed to remove the potential for column name
+        # conflicts for elements not needed in the query.
+        if filter_to_specs:
```

Comment:
Don't we need to do this selection whenever there are `where_filter_specs` instead of just when `filter_to_specs` is set?

Reply:
This will only ever be empty in the case of a nested aggregation (i.e., nested derived offset metrics) so we've already removed all the unnecessary specs in those cases.
There's also a totally unrelated bug in that code that might necessitate a change here but this retains the current behavior.